src/panels/lovelace/cards/hui-glance-card.ts

```diff
@@ -85,6 +86,28 @@ export class HuiGlanceCard extends hassLocalizeLitMixin(LitElement)
     }
   }
 
+  public getElementConfig(config: any, hass: HomeAssistant): TemplateResult {
```

Comment:
We should return a promise that returns to `HTMLElement`.

Why? That way we can:
 - Start code splitting the config elements. We don't need to load this unless the user has a glance card and is editing it (not common)
 - If it's a generic element, this class could configure it.

Reply:
We can use `until` directive to render a spinner until it's loaded https://polymer.github.io/lit-html/guide/writing-templates#untilpromise-defaultcontent

Reply:
code would look something like this:

```ts
// LovelaceCardEditor is a new interface for Lovelace cards
public async getElementConfig(): Promise<LovelaceCardEditor> {
  await import('../editors/hui-glance-card-editor');
  return document.createElement('hui-glance-card-editor');
}
```

Reply:
With `hui-glance-card-editor` just containing a card as usual.

Reply:
It inherits from HTMLElement, however, we can always implement interfaces that extend HTMLElement, like we did with `LovelaceCard`.

Reply:
It was complaining even though I had done this but its all good now...

Reply:
I need help with making this a promise and using the setconfig/setHass/adding event listener to the LovelaceCardEditor element result

Reply:
```ts
async getElementConfig(config, hass) {
  const cardClass = …
  const configEl = await cardClass.getElementConfig()
  configEl.setConfig(config);
  configEl.hass = hass;
}
```